### PR TITLE
[DOCS][MINOR] Fixed a few typos in the Structured Streaming documentation

### PR DESCRIPTION
**Pull request by aseigneurin**
Fixed a few typos.

There is one more I'm not sure of:

```
        Append mode uses watermark to drop old aggregation state. But the output of a 
        windowed aggregation is delayed the late threshold specified in `withWatermark()` as by
        the modes semantics, rows can be added to the Result Table only once after they are
```

Not sure how to change `is delayed the late threshold`.

### CODE REVIEW

File: docs/structured-streaming-programming-guide.md [hunk 1]
```diff
@@ -713,15 +713,15 @@ old windows correctly, as illustrated below.
 
 ![Handling Late Data](img/structured-streaming-late-data.png)
 
-However, to run this query for days, it's necessary for the system to bound the amount of 
+However, to run this query for days, it is necessary for the system to bound the amount of
```
Comment (srowen) [inline docs/structured-streaming-programming-guide.md]: Not a typo

File: docs/structured-streaming-programming-guide.md [hunk 2]
```diff
@@ -930,7 +930,7 @@ There are a few types of output modes.
 new rows added to the Result Table since the last trigger will be 
 outputted to the sink. This is supported for only those queries where 
 rows added to the Result Table is never going to change. Hence, this mode 
-guarantees that each row will be output only once (assuming 
+guarantees that each row will be outputted only once (assuming
```
Comment (srowen) [inline docs/structured-streaming-programming-guide.md]: Not a typo

**Review by srowen — CHANGES_REQUESTED** [review]
A couple should be reverted but the rest are OK

File: docs/structured-streaming-programming-guide.md (hunk 2 above)
Comment (aseigneurin) [inline docs/structured-streaming-programming-guide.md]: Not a typo, I agree, but I thought it made sense to keep the same formulation as what is used in the same paragraph

Comment (srowen) [inline docs/structured-streaming-programming-guide.md]: Oh, hm. As a native speaker, I would have said "outputted" is incorrect actually. But it looks like it's an accepted alternative. Well, whatever's the smaller change to achieve consistency.

**Comment by srowen** [comment]
@tdas what did you intend in the paragraph cited in the description? I'm also not sure.

File: docs/structured-streaming-programming-guide.md (hunk 2 above)
Comment (aseigneurin) [inline docs/structured-streaming-programming-guide.md]: Interesting! I also thought "outputted" was strange.

I have reverted my change anyway.

**Comment by srowen** [comment]
@aseigneurin Please read http://spark.apache.org/contributing.html too. As you might imagine "typos" isn't useful as a title. Write "[MINOR]" too

**Comment by aseigneurin** [comment]
@srowen does this mean I should open a JIRA for this?

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**